dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -293,6 +310,8 @@ def _collect_relations(self, dynatrace_component, external_id, component_type):
                     if component_type == 'custom-device':
                         source_relation_id = source_id.get('id')
                         self.relation(source_relation_id, external_id, relation_type, {})
+                    elif component_type == 'monitor':
```

Comment:
Also this block should be refactored when we create unit test for all relation. Lines 293-304 and 305-316 have too much similar code, this bit should be extracted to its own function.